### PR TITLE
Add lobbytrack reports

### PR DESCRIPTION
**Pull request by jgreben**
In order to be able to perform the controller tests (that double the TinyTDS gem) it is necessary that `freetds` be installed on Travis. (Not that happy about this.)

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -51,6 +51,8 @@ gem 'has_scope'
 gem 'tsv'
 # Rails javascript runtime environment
 gem 'therubyracer'
+# To query an sqlserver instance (lobbytrack)
+gem 'tiny_tds'
```
Comment (shelleydoljack) [inline Gemfile]: Should we pin this to a specific version that corresponds to the version of freetds that we have on travis and the libsys-webforms server?

File: app/models/ability.rb [hunk 2]
```diff
@@ -45,9 +45,10 @@ def assign_staff_read_permission(current_user)
 
   def assign_staff_manage_permission(current_user)
     can :manage, AccessionNumberUpdate if /A|Y/.match?(current_user.accession_number)
-    can :manage, ManagementReport if /A|Y/.match?(current_user.mgt_rpts)
     can :manage, EdiInvoice if /A|Y/.match?(current_user.edi_inv_manage)
     can :manage, EdiErrorReport if /A|Y/.match?(current_user.edi_inv_manage)
+    can :manage, LobbytrackReport if /A|Y/.match? current_user.lobbytrack_report
```
Comment (shelleydoljack) [inline app/models/ability.rb:50]: If access is restricted by IP, is this necessary? Also, are the `:manage` abilities what we want if the routes file shows only `:index`? Or do the post actions in the routes file equate to `:manage` here?

File: README.md [hunk 3]
```diff
@@ -97,3 +97,18 @@ is no Rails ORM on which to base an Ability.)  Clicking the "Run" link will exec
 already configured OCI8 connection gem (using the environment's database.yml connection details). If more jobs need to be
 added in the future, just fill out a new section under the pl_sql_jobs section in the https://github.com/sul-dlss/shared_configs
 repository and follow the instructions there for deploying to the application server.
+
+## Lobbytrack Report
+
+ The lobbytrack database connection settings are kept in the Settings config file along with an array of IP addresses that 
+have firewall access to the lobbytrack database and can also use this webform. Inclusion of the client IP address here will 
+allow the Lobbytrack report menu item to appear for only cliets with the IPs listed here:
```
Comment (shelleydoljack) [inline README.md]: Typo here: `cliets`.

File: README.md [hunk 4]
```diff
@@ -97,3 +97,18 @@ is no Rails ORM on which to base an Ability.)  Clicking the "Run" link will exec
...
+```yml
+# Lobbytrack settings
+lobbytrack_host: '-----'
+lobbytrack_user: '-----'
+lobbytrack_password: '-----'
+lobbytrack_db: '-----'
+lobbytrack_port: -----
+lobbytrack_ips: ['1.1.1.1', '2.2.2.2']
```
Comment (shelleydoljack) [inline README.md:113]: I get a rails error when I try to run the server locally. If I add the above to my `config/settings/development.yml` file, I don't get an error, but I am also not able to view any of the Lobbytrack report pages. How do we get this to work locally?

**Review by shelleydoljack — COMMENTED** [review]
Let's walk through this PR together. Looks great. I have some questions about how to run it locally.

File: Gemfile (hunk 1 above)
Comment (jgreben) [reply Gemfile]: That's a good idea.

File: app/models/ability.rb (hunk 2 above)
Comment (jgreben) [reply app/models/ability.rb:50]: The IP of the lobbytrack machine is a general use staff machine that is used by multiple people throughout the day, so I felt it was also necessary to further restrict the access.

Yes, `:manage` is for the various post actions. It seems kind of linguistically awkward to have something like `can %i[visits checkins visit_dates checkin_dates]`... and a bit cumbersome to define a bunch of alias_actions. I suppose the method names could be changed to something more verby but then the file names of the views, etc. see kind of weird..

As an aside: Looking at the Ability model again, and the `:manage` action it seems that 
`alias_action :activate, :deactivate, to: :manage` (for package management?) might be unnecessary..

File: README.md (hunk 4 above)
Comment (jgreben) [reply README.md:113]: We would have to add your hardline IP address to the LobbyTrack firewall rules and then you would be able to run it by starting up your localhost server like this: `REMOTE_USER=sdoljack` rails s -b 177.66.555.44

Let me know yout IP and I will put in the request.

**Review by jgreben — COMMENTED** [review]
See comments.

**Review by shelleydoljack — APPROVED** [review]

**Pull request merged**